dali/test/python/test_video_pipeline.py

```diff
@@ -273,3 +272,53 @@ def check_container(cont):
 def test_container():
     for cont in video_types:
         yield check_container, cont
+
+def test_pad_sequence():
+    @pipeline_def(batch_size=1, num_threads=4, device_id=0)
+    def create_video_pipe(filenames, sequence_length=1, stride=1, step=-1, pad_sequences=False):
+        fr, lab, fr_num, time_stamp = fn.readers.video(device="gpu", filenames=filenames, labels=[],
+                                                       sequence_length=sequence_length,
+                                                       shard_id=0, num_shards=1, enable_timestamps=True,
+                                                       enable_frame_num=True, random_shuffle=False,
+                                                       skip_vfr_check=True, step=step, stride=stride,
+                                                       pad_last_batch=True, pad_sequences=pad_sequences)
+        return fr, lab, fr_num, time_stamp
+
+    video_filename = [os.path.join(video_data_root, 'sintel', 'video_files', 'sintel_trailer-720p_3.mp4')]
+    dali_pipe = create_video_pipe(video_filename)
+    dali_pipe.build()
+    meta = dali_pipe.reader_meta()
+    total_number_of_frames = list(meta.values())[0]['epoch_size']
+
+    sequence_length = 4
+    stride = sequence_length//2
+    # second sequence should have only half of the frames
+    step = total_number_of_frames - (stride * sequence_length//2 - 1)
+    dali_pipe = create_video_pipe(batch_size=2, filenames=video_filename, sequence_length=sequence_length,
+                                  stride=stride, step=step, pad_sequences=True)
+    dali_pipe.build()
+    meta = dali_pipe.reader_meta()
+    assert list(meta.values())[0]['epoch_size'] == 2
+
+    last_sample_frame_count = 1 + (total_number_of_frames - 1 - step) // stride
+    assert last_sample_frame_count < sequence_length
+
+    out = dali_pipe.run()
+    sampl = 1
+    # check padded sample
+    # non padded frames should not be 0
+    assert np.any(np.array(out[0].as_cpu()[sampl])[0:last_sample_frame_count]) != 0
+    # while padded one only 0
+    assert np.all(np.array(out[0].as_cpu()[sampl])[last_sample_frame_count + 1:]) == 0
+    assert np.array(out[2].as_cpu()[sampl]) == step
+    # non padded samples should have non negative timestamps
+    assert np.all(np.array(out[3].as_cpu()[sampl])[0:last_sample_frame_count] != np.array([-1] * last_sample_frame_count))
+    # while padded one only -1
+    assert np.all(np.array(out[3].as_cpu()[sampl])[last_sample_frame_count + 1:] == np.array([-1] * (sequence_length - last_sample_frame_count)))
+
+    dali_pipe = create_video_pipe(batch_size=2, filenames=video_filename, sequence_length=sequence_length,
+                                  stride=stride, step=step, pad_sequences=False)
+    dali_pipe.build()
+    meta = dali_pipe.reader_meta()
+    # when sequence padding if off we should get only one valid sample in the epoch
+    assert list(meta.values())[0]['epoch_size'] == 1
```

Comment:
Reading this I had an idea that maybe it would be useful to add second pipeline to this test with `batch_size=1` and check that all frames are still there. This is just a thought, I'm totally fine with how it is now.

Reply:
Will try.

Reply:
Done